hazelcast/compact.py

```diff
@@ -1,31 +1,46 @@
 import logging
 import typing
 
-from hazelcast.errors import HazelcastSerializationError
+from hazelcast.errors import HazelcastSerializationError, IllegalStateError
 from hazelcast.future import Future, ImmediateFuture
-from hazelcast.invocation import InvocationService, Invocation
+from hazelcast.invocation import Invocation
 from hazelcast.protocol.codec import (
     client_fetch_schema_codec,
     client_send_schema_codec,
     client_send_all_schemas_codec,
 )
-from hazelcast.serialization.compact import (
-    CompactStreamSerializer,
-    Schema,
-    SchemaNotReplicatedError,
-)
+
+if typing.TYPE_CHECKING:
+    from hazelcast.cluster import ClusterService
+    from hazelcast.config import Config
+    from hazelcast.invocation import InvocationService
+    from hazelcast.protocol.client_message import OutboundMessage
+    from hazelcast.reactor import AsyncoreReactor
+    from hazelcast.serialization.compact import (
+        CompactStreamSerializer,
+        Schema,
+        SchemaNotReplicatedError,
+    )
 
 _logger = logging.getLogger(__name__)
 
 
 class CompactSchemaService:
+    _SEND_SCHEMA_RETRY_COUNT = 100
```

Comment:
100 is an arbitrary value that we believed would be a nice default. It is an internal implementation detail (the property you see there is not public) but we wanted to give a way to configure it if someone really wants (through private APIs like this). So, I believe we don't have to use the same name. We use put there because the method to send schemas is called put there, but not here.